great_expectations/datasource/fluent/sources.py

```diff
@@ -261,7 +261,7 @@ def _bind_asset_factory_method_if_not_present(
         asset_type_name: str,
     ):
         add_asset_factory_method_name = f"add_{asset_type_name}_asset"
-        asset_factory_defined: bool = add_asset_factory_method_name in ds_type.__dict__
+        asset_factory_defined: bool = hasattr(ds_type, add_asset_factory_method_name)
```

Comment:
This change is necessary since `__dict__` doesn't look up inheritance tree while `hasattr` will. I noticed that the sql `add_*_asset` code wasn't being called and that's because this dynamically defined function was overwriting it.